http3/server.go

```diff
@@ -479,6 +492,24 @@ func (s *Server) handleConn(conn quic.Connection) error {
 		}
 	}
 
+	s.connCount.Add(1)
+
+	var (
+		lastID atomic.Int64
+	)
+	// first valid ID is zero, subtract 4, so goaway frame id starts at 0
+	lastID.Store(-4)
+	// run in a separate goroutine
+	stop := context.AfterFunc(s.closeCtx, func() {
+		b = (&goawayFrame{
+			StreamID: quic.StreamID(lastID.Load()) + 4,
+		}).Append(b[:0])
+		str.Write(b)
+	})
```

Comment:
Every function is going to get run a separate Go routine, leading a large number of spawned Go routines when the server is shut down.

What do you think of doing this in the the `for` loop that accepts streams? We'd pass the `closeCtx` to `hconn.acceptStream`, and introduce a separate check when this function returns an error.

Reply:
Done, and simpler I think